src/ripple/app/tx/impl/CreateOffer.cpp

```diff
@@ -1107,6 +1107,12 @@ CreateOffer::applyGuts(Sandbox& sb, Sandbox& sbCancel)
     if (bImmediateOrCancel)
     {
         JLOG(j_.trace()) << "Immediate or cancel: offer canceled";
+        if (!crossed && sb.rules().enabled(featureImmediateOfferKilled))
+            // If the ImmediateOfferKilled amendment is enabled, any
+            // ImmediateOrCancel offer that transfers absolutely no funds
+            // returns tecKILLED rather than tesSUCCESS.  Motivation for the
+            // change is here: https://github.com/ripple/rippled/issues/4115
+            return {tecKILLED, false};
```

Comment:
This is broken: it basically changes the semantics of **`Immediate or Cancel`** to be those of **`Fill or Kill`**.

A "Fill or Kill" order means: "if the offer can't be entirely filled by crossing against existing offers, abort. Never place an order on the order book."

An "Immediate or Cancel" order, on the other hand, means: "fill as much as you can by crossing against existing offers. If the offer isn't fully filled, that's fine, cancel the rest and never put an offer on the book."

So returning `false` from here is a bug. It should return `true` because you **want** to apply the executed crossings, but you don't want to go any further (i.e. to place the offer on the book). 

Given the difference between FoK and IoC, I think that `tecKILLED` is the wrong error code here; it should be `tecPARTIALLY_FILLED` or somesuch.

Reply:
@nbougais, are you _sure_ this is broken?  Note the check for `!crossed` in the conditional.  So we should `return {tecKILLED, false}` only in the case where _no_ funds are transferred.  If _any_ funds were transferred, then we continue to `return {tesSUCCESS, true}` as before.  At least that's what I read `crossed` as representing.

What am I missing?